dev/com.ibm.ws.concurrent.persistent_fat_multiple/fat/src/com/ibm/ws/concurrent/persistent/fat/multiple/DatabaseContainerFactory.java

```diff
@@ -0,0 +1,144 @@
+package com.ibm.ws.concurrent.persistent.fat.multiple;
+
+import java.time.Duration;
+
+import org.testcontainers.containers.Db2Container;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+import org.testcontainers.containers.MockServerContainer;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.containers.MSSQLServerContainer;
+import org.testcontainers.containers.OracleContainer;
+import org.testcontainers.containers.output.OutputFrame;
+
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.custom.junit.runner.FATRunner;
+
+/**
+ * This is a factory class that creates a database test container.
+ * The test container returned will be based on the 'fat.bucket.db.type' 
+ * system property. 
+ * 
+ * The 'fat.bucket.db.type' property is set to different databases
+ * by our test infrastructure when a fat-suite has the property
+ * 'fat.test.databases' set to true
+ * 
+ * TODO: Since this factory class is general can it be moved to fattest simplicity?
+ *
+ */
+public class DatabaseContainerFactory {
+	private static String database = System.getProperty("fat.bucket.db.type", "Derby");
+	private static GenericContainer<?> container = initContainer();
+	
+	@SuppressWarnings("rawtypes")
+	private static Class[] classes;
+	
+	/**
+	 * Reads the 'fat.bucket.db.type' system property and returns a
+	 * container based on that property.
+	 * 
+	 * If we are using a network database, a test container running
+	 * that database will be returned.  
+	 * [Postgre, DB2, Oracle, SQLServer, MySQL]
+	 * 
+	 * If we are using derby embedded or unsupported database
+	 * a no-op container will be returned.
+	 * [Derby, Informix, Sybase]
+	 * 
+	 * @param testClasses - Provide the name(s) of your test class(es). These will be passed to the test container which will then provide log output to the class(es) log.  
```

Comment:
I don't think it's necessary to pass in the test classes just for the sake of logging. I recommend either:
A) Let the FAT bucket set up its own logging by calling `withLogConsumer()` on the returned `GenericContainer`
B) do the logging as `DatabaseContainerFactory.class` instead of as the test class
I have a slight preference towards (B) since it's less boiler plate for the FAT to worry about (once we move this all to fattest.simplicity)